test/mpi/gentests.sh

```diff
@@ -0,0 +1,162 @@
+#!/bin/sh
+#
+# (C) 2018 by Argonne National Laboratory.
+#     See COPYRIGHT in top-level directory.
+#
+# Generate datatype testing build instructions for every line in
+# configuration file dtp-test-config.txt
+
+types=""     # basic MPI datatypes for type signature
+counts=""    # counts for type signature
+pathname=""  # test file pathname
+testdir=""   # name of test directory
+testname=""  # name of test
+builddir=""  # directory in which executable will be located
+dirs=""      # all directories containing DTPools tests
+args=""      # extra args
+timelimit="" # time limit for testlist.in
+procs=""     # number of processes to use in testlist.in
+srcdir=""    # relative directory in which this script is located
+maxbufsize=""
+other_args=""
+
+# help message
+show_help() {
+cat << EOF
+Usage: $0 [options]
+options:
+  --help    show this message and exit
+  --with-dtpools-datatypes=[typelist]
+            comma separated list of MPI Datatypes to use for
+            generating DTPools tests
+  --with-dtpools-maxbufsize=[bytes]
+            max buffer size of DTP objects
+EOF
+}
+
+while [ true ] ; do
+    if test -z "$1" ; then break ; fi
+    case $(echo $1 | cut -d= -f1) in
+        --help)
+            show_help
+            exit 1
+            ;;
+        --with-dtpools-datatypes)
+            types=$(echo $1 | cut -d= -f2)
+            shift
+            ;;
+        --with-dtpools-maxbufsize)
+            maxbufsize=$(echo $1 | cut -d= -f2)
+            shift
+            ;;
+    esac
+done
+
+# get source and build directories
+srcdir=`dirname $0`
+builddir=$PWD
+
+# reformat type list
+types=$(echo $types | tr "," " ")
+
+# set max count and large count
+MAXSIZE=1000
+MAXPINGSIZE=100
+MAXCOUNT=262144
```

Comment:
These three variables are no longer needed. I can remove them if you want.

Reply:
These have been removed in a later fixup commit.